nixos/tests/installer.nix

```diff
@@ -232,25 +233,29 @@ let
 
             # The test cannot access the network, so any packages we
             # need must be included in the VM.
-            system.extraDependencies = with pkgs;
-              [ sudo
-                libxml2.bin
-                libxslt.bin
-                docbook5
-                docbook5_xsl
-                unionfs-fuse
-                ntp
-                nixos-artwork.wallpapers.gnome-dark
-                perlPackages.XMLLibXML
-                perlPackages.ListCompare
-                xorg.lndir
-
-                # add curl so that rather than seeing the test attempt to download
-                # curl's tarball, we see what it's trying to download
-                curl
-              ]
-              ++ optional (bootLoader == "grub" && grubVersion == 1) pkgs.grub
-              ++ optionals (bootLoader == "grub" && grubVersion == 2) [ pkgs.grub2 pkgs.grub2_efi ];
+            system.extraDependencies = let
+              # This is because the EFI version of grub overrides the base
+              # version of GRUB 2, so we can't just use pkgs.grub2_efi here.
+              grub2 = pkgs.grub2_light.override { zfsSupport = grubUseZfs; };
```

Comment:
Should probably use if grubUseZfs then pkgs.grub2 else pkgs.grub2_light, and move the comment about the EFI stuff to below this line.

Reply:
@grahamc: It's already a bit obscure that way, doing `if grubUseZfs then pkgs.grub2 else pkgs.grub2_light` IMHO makes it worse because then you have to look at the actual definition in `all-packages.nix` to check which flags are enabled (which itself again uses `.override`).

Reply:
I guess my perspective is the `_light` variant becomes meaningless and overriding an overridden package to disable the override seems really weird.

Reply:
I agree. Ideally the `grub2` package should be the main package and `grub2_full` should be an override of `grub2_light` with all that stuff enabled. In the line here `pkgs.grub2.override` would also work, because `grub2_full` is just `grub2_light` with ZFS support.